### DISCUSSION TITLE
[FEATURE]How to customize checks.

### DISCUSSION THREAD

**Discussion by geeeez** [schemathesis/schemathesis#1443] [Q&A]
schemathesis has four built-in checks.However, there is no information on conventional web vulnerabilities.If I want to customize my own check,Works like a built-in check，What should I start with? Do I need to know the whole workflow of schemathesis？Is there an interface that allows users to customize their own check, such as scanning SQL injection or logic ultra vires？Is there a way to make the check defined by me work like the built-in check.
<img width="723" alt="image" src="https://user-images.githubusercontent.com/29922543/160076706-dcd7e7c5-be60-48a3-860d-75eb486a963a.png">

**Comment by Stranger6667** [comment]
Hi! Sorry for not responding earlier.

I think that these chapters([1](https://schemathesis.readthedocs.io/en/stable/extending.html#checks), [2](https://schemathesis.readthedocs.io/en/stable/cli.html#registering-custom-checks)) of our docs should give some overview on that. You'll be able to write a check that works as a built-in one. 

Though, they indeed do not provide any info on how Schemathesis works (#1085 should describe this). Generally, a check is executed each time Schemathesis receives the app's response in CLI - in Python tests, you have to call `case.call_and_validate` / `case.validate_response`.

>  such as scanning SQL injection or logic ultra vires？

I think, such a feature goes to both, data generation + a check. First, you need to generate a proper data sample, e.g. includes an SQL injection (you can use [these hooks](https://schemathesis.readthedocs.io/en/stable/extending.html#before-generate) for it), then write a check that verifies whether the injection was successful or not.

I hope, that those comments help at least a little bit. Let me know if you'd need any further elaboration
